src/catchup/CatchupRange.cpp

```diff
@@ -0,0 +1,40 @@
+// Copyright 2017 Stellar Development Foundation and contributors. Licensed
+// under the Apache License, Version 2.0. See the COPYING file at the root
+// of this distribution or at http://www.apache.org/licenses/LICENSE-2.0
+
+#include "catchup/CatchupRange.h"
+#include <utility>
+
+namespace stellar
+{
+
+CatchupRange::CatchupRange(CheckpointRange range, uint32_t stopAt,
+                           bool applyBuckets)
+    : mRange{std::move(range)}, mStopAt{stopAt}, mApplyBuckets{applyBuckets}
+{
```

Comment:
no validation for mStopAt?

Reply:
Not sure how to do that, It should be in range of [range.last() - checkpoint_size + 1...range.last()] but checkpoint_size is user-configurable. Also passing HistoryManager to that constructor does not seem to be a good idea.

Do you have any suggestions?

Reply:
it sounds like having checkpoint member of `CheckpointRange` would solve this. In the previous PR I thought we would only use it for validation of `CheckpointRange`, but it makes sense to remember it in that structure